rct_optimizations/test/src/pose_generator.cpp

```diff
@@ -0,0 +1,99 @@
+#include "rct_optimizations_tests/observation_creator.h"
+#include "rct_optimizations_tests/pose_generator.h"
+#include <stdexcept>
+
+// hemisphere
+
+std::vector<Eigen::Isometry3d>
+rct_optimizations::test::genHemispherePose(Eigen::Isometry3d target_pose, double r, int theta_cnt, int phi_cnt)
+{
+  // all in the target coordinate system, with z forward
+  std::vector<Eigen::Isometry3d> camera_positions;
+
+  Eigen::VectorXd theta_range = Eigen::VectorXd::LinSpaced(theta_cnt, 0.0, M_PI);
+  Eigen::VectorXd phi_range = Eigen::VectorXd::LinSpaced(phi_cnt, 0.0, M_PI);
+
+  for (long theta_it = 0; theta_it < theta_range.size(); ++theta_it)
```

Comment:
Being picky, but `std::size_t` is a better choice than `long` because it's unsigned